### PR TITLE
✨Add COM Energiemix

### PR DESCRIPTION
**Pull request by hf-aschloegl**
Fixes https://github.com/Hochfrequenz/BO4E-python/issues/158

### CODE REVIEW

File: src/bo4e/com/energiemix.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+@attr.s(auto_attribs=True, kw_only=True)
+class Energiemix(COM):
+    """
+    Zusammensetzung der gelieferten Energie aus den verschiedenen Primärenergieformen.
+    """
+
+    # required attributes
+    #: Eindeutige Nummer zur Identifizierung des Energiemixes.
+    energiemixnummer: int = attr.ib(validator=attr.validators.instance_of(int))
+    #: Strom oder Gas etc..
```
Comment (hf-kklein) [inline src/bo4e/com/energiemix.py:30]:
```suggestion
    #: Strom oder Gas etc…
```

File: src/bo4e/com/energiemix.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+# pylint: disable=too-few-public-methods, too-many-instance-attributes
+@attr.s(auto_attribs=True, kw_only=True)
+class Energiemix(COM):
+    """
+    Zusammensetzung der gelieferten Energie aus den verschiedenen Primärenergieformen.
+    """
+
+    # required attributes
+    #: Eindeutige Nummer zur Identifizierung des Energiemixes.
```
Comment (hf-kklein) [inline src/bo4e/com/energiemix.py:28]:
```suggestion
    #: Eindeutige Nummer zur Identifizierung des Energiemixes
```

File: src/bo4e/com/energiemix.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    energiemixnummer: int = attr.ib(validator=attr.validators.instance_of(int))
+    #: Strom oder Gas etc..
+    energieart: Sparte = attr.ib(validator=attr.validators.instance_of(Sparte))
+    #: Bezeichnung des Energiemix.
```
Comment (hf-kklein) [inline src/bo4e/com/energiemix.py:32]:
```suggestion
    #: Bezeichnung des Energiemix
```

File: src/bo4e/com/energiemix.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    bezeichnung: str = attr.ib(validator=attr.validators.instance_of(str))
+    #: Jahr, für das der Energiemix gilt.
```
Comment (hf-kklein) [inline src/bo4e/com/energiemix.py:34]:
```suggestion
    #: Jahr, für das der Energiemix gilt
```

File: src/bo4e/com/energiemix.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    gueltigkeitsjahr: int = attr.ib(validator=attr.validators.instance_of(int))
+    #: Anteile der jeweiligen Erzeugungsart.
```
Comment (hf-kklein) [inline src/bo4e/com/energiemix.py:36]:
```suggestion
    #: Anteile der jeweiligen Erzeugungsart
```

File: src/bo4e/com/energiemix.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    #: Höhe des erzeugten Atommülls in g/kWh.
+    atommuell: Decimal = attr.ib(default=None)
+    #: Zertifikate für den Energiemix.
+    oekozertifikate: List[Oekozertifikat] = attr.ib(default=[])
+    #: Ökolabel für den Energiemix.
+    oekolabel: List[Oekolabel] = attr.ib(default=[])
+    #: Kennzeichen, ob der Versorger zu den Öko Top Ten gehört.
+    oeko_top_ten: bool = attr.ib(default=None)
+    #: Internetseite, auf der die Strommixdaten veröffentlicht sind.
+    website: str = attr.ib(default=None)
```
Comment (hf-kklein) [inline src/bo4e/com/energiemix.py:53]:
```suggestion
    #: Bemerkung zum Energiemix
    bemerkung: str = attr.ib(default=None)
    #: Höhe des erzeugten CO2-Ausstosses in g/kWh
    co2_emission: Decimal = attr.ib(default=None)
    #: Höhe des erzeugten Atommülls in g/kWh
    atommuell: Decimal = attr.ib(default=None)
    #: Zertifikate für den Energiemix
    oekozertifikate: List[Oekozertifikat] = attr.ib(default=[])
    #: Ökolabel für den Energiemix
    oekolabel: List[Oekolabel] = attr.ib(default=[])
    #: Kennzeichen, ob der Versorger zu den Öko Top Ten gehört
    oeko_top_ten: bool = attr.ib(default=None)
    #: Internetseite, auf der die Strommixdaten veröffentlicht sind
    website: str = attr.ib(default=None)
```

File: src/bo4e/com/energiemix.py [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    """
+    Schema for de-/serialization of Energiemix.
+    """
+
+    # required attributes
+    energiemixnummer = fields.Int()
+    energieart = EnumField(Sparte)
+    bezeichnung = fields.Str()
+    gueltigkeitsjahr = fields.Int()
+    anteil = fields.List(fields.Nested(EnergieherkunftSchema))
```
Comment (hf-kklein) [inline src/bo4e/com/energiemix.py:66]: minimal length 1

**Review by hf-kklein — APPROVED** [review]

**Review by hf-krechan — APPROVED** [review]

File: src/bo4e/com/energiemix.py (hunk 1 above)
Comment (hf-aschloegl) [reply src/bo4e/com/energiemix.py:30]:
```suggestion
    #: Strom oder Gas etc.
```

File: src/bo4e/com/energiemix.py (hunk 7 above)
Comment (hf-aschloegl) [reply src/bo4e/com/energiemix.py:75]: a7bee14cb46f59de62952f19fd04eb01046c8da0

**Pull request merged**